dspace-api/src/main/java/org/dspace/content/CollectionServiceImpl.java

```diff
@@ -907,4 +919,66 @@ public Group createDefaultReadGroup(Context context, Collection collection, Stri
         return role;
     }
 
+    @Override
+    public List<Collection> findAuthorizedCollectionsInSOLR(String q, Context context, Community community,
+        int offset, int limit) throws SQLException, SearchServiceException {
+
+        List<Collection> collections = new ArrayList<Collection>();
+        StringBuilder query = new StringBuilder();
```

Comment:
This StringBuilder object is created here, but isn't used until the resultSolrQuery() method. Can you move the creation of the variable to the resultSolrQuery() method.